### PR TITLE
Add Storage upload inputStream doc

### PR DESCRIPTION
**Pull request by changxu0306**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/storage/fragments/android/upload.md [hunk 1]
```diff
@@ -136,3 +137,60 @@ upload
 
 </amplify-block>
 </amplify-block-switcher>
+
+## Upload inputStream
```
Comment (richardmcclellan) [inline docs/lib/storage/fragments/android/upload.md:141]: Let's put this one _above_ `uploadFile`, as it is the more commonly used one on Android.  iOS is also similar - they have `uploadData` (equivalent of `uploadInputStream`) above `uploadFile`

File: docs/lib/storage/fragments/android/upload.md [hunk 2]
```diff
@@ -136,3 +137,60 @@ upload
...
+To upload to S3 from a data object, specify the key and the InputStream to be uploaded. 
+
+<amplify-block-switcher>
+<amplify-block name="Java">
+
+```java
+private void uploadInputStream() {
+    InputStream exampleInputStream = new ByteArrayInputStream(RandomBytes.bytes());
```
Comment (richardmcclellan) [inline docs/lib/storage/fragments/android/upload.md:149]: For the sample, can you generate how to obtain the `InputStream` from a `Uri`?  A `Uri` is what you get back from the Android file/photo picker, so this is the most common use case that customers will be needing.

File: docs/lib/storage/fragments/android/upload.md [hunk 3]
```diff
@@ -136,3 +137,60 @@ upload
 
 </amplify-block>
 </amplify-block-switcher>
+
+## Upload inputStream
+To upload to S3 from a data object, specify the key and the InputStream to be uploaded.
```
Comment (richardmcclellan) [inline docs/lib/storage/fragments/android/upload.md:142]:
```suggestion
To upload to S3 from a data object, specify the key and the `InputStream` to be uploaded. 
```

File: docs/lib/storage/fragments/android/upload.md [hunk 4]
```diff
@@ -1,3 +1,61 @@
+## Upload inputStream
+To upload to S3 from a data object, specify the key and the `InputStream` to be uploaded.
```
Comment (jamesonwilliams) [inline docs/lib/storage/fragments/android/upload.md:2]:
```suggestion
Amplify allows you to upload `File`s and `InputStream`s.

## Upload InputStream
To upload data to S3 from an `InputStream`:
```

File: docs/lib/storage/fragments/android/upload.md [hunk 5]
```diff
@@ -1,3 +1,61 @@
...
+
+```kotlin
+private fun uploadInputStream() {
+    val exampleInputStream = getContentResolver().openInputStream(uri)
+
+    Amplify.Storage.uploadInputStream(
+        "ExampleKey",
+        exampleInputStream,
+        { result-> Log.i("MyAmplifyApp", "Successfully uploaded: " + result.getKey()) },
+        { storageFailure-> Log.e("MyAmplifyApp", "Upload failed", storageFailure) }
```
Comment (jamesonwilliams) [inline docs/lib/storage/fragments/android/upload.md:31]:
```suggestion
        { Log.i("MyAmplifyApp", "Successfully uploaded: ${it.getKey()}.") },
        { Log.e("MyAmplifyApp", "Upload failed", it) }
```

**Review by jamesonwilliams — APPROVED** [review]

**Pull request merged**